server/ladder_service.py

```diff
@@ -501,6 +518,32 @@ async def shutdown(self):
             queue.shutdown()
 
 
+def parse_map_params(map_params: str, weight: int) -> NeroxisGeneratedMap:
```

Comment:
Could be a class method of `NeroxisGeneratedMap`?

Reply:
It could be but wasn't sure since the type is in the params itself. and if another type was introduced it wouldn't be part of the neroxis method

Reply:
IMO it makes sense to check the type first and then delegate to the appropriate method/function.

Reply:
I'm just thinking that this function is basically a constructor, so it would make sense for it to be a class method on the class it's constructing.

Reply:
Aight